contracts/AelinUpFrontDeal.sol

```diff
@@ -397,8 +400,13 @@ contract AelinUpFrontDeal is AelinERC20, MinimalProxyFactory, IAelinUpFrontDeal
         amountVested[msg.sender] += underlyingDealTokensClaimed;
         _burn(msg.sender, underlyingDealTokensClaimed);
         totalUnderlyingClaimed += underlyingDealTokensClaimed;
-        IERC20(_underlyingDealToken).safeTransfer(msg.sender, underlyingDealTokensClaimed);
-        emit ClaimedUnderlyingDealToken(msg.sender, _underlyingDealToken, underlyingDealTokensClaimed);
+        // This could potentially be the case where the last user claims a slightly smaller amount if there is some precision loss
+        // although it will generally never happen as solidity rounds down so there should always be a little bit left
+        uint256 precisionAdjustedClaim = underlyingDealTokensClaimed > IERC20(_underlyingDealToken).balanceOf(address(this))
```

Comment:
also we should probably put the precision adjustment in the view function and then pass that value down the more I think about it. can you make that change while you are working on the tests. We need to be considerate about the amountVested section but I think its probably better. can you take a look at this and decide what you think please? I will look at it again Monday too

Reply:
if we do it that way in the view instead clients will always have the correct amount when they ask the contract for how much you get